src/o365/spfx/commands/project/project-upgrade.ts

```diff
@@ -232,17 +295,55 @@ class SpfxProjectUpgradeCommand extends Command {
         f.resolution = f.resolution.replace('install', this.getPackageManagerCommand('install'));
         return;
       }
+
+      //copy support for multiple shells
+      if (f.resolution.startsWith('copy')) {
+        f.resolution = f.resolution.replace('copy', this.getCopyCommand('copyCommand'));
+        f.resolution = f.resolution.replace('DestinationParam', this.getCopyCommand('copyDestinationParam'));
+        return;
+      }
+      //createdir support for multiple shells
+      if (f.resolution.startsWith('createDir')) {
+        f.resolution = f.resolution.replace('createDir', this.getDirectoryCommand('createDirectoryCommand'));
+        f.resolution = f.resolution.replace('NameParam', this.getDirectoryCommand('createDirectoryNameParam'));
+        f.resolution = f.resolution.replace('ItemTypeParam', this.getDirectoryCommand('createDirectoryItemTypeParam'));
+        return;
+      }
+      //'Add' support for multiple shells
+      if (f.resolution.startsWith('Add')) {
```

Comment:
@waldekmastykarz , @BassemNKhalil  quick question. 

Should we make that `Add` token/text more prominent something like `##_Add_##` OR [Add] so we would be sure that only rules that have that token/text will be replaced and not other rules resolutions will be impacted. 

For example is someone not knowing about that implementation here adds a rule that has resolution of a kind `Add text at line 24 in file N` then this piece of code will try to find `'[BEFOREPATH]'` for that particular rule and fail.

 Same applies for words like 'copy', 'Remove' even 'install'

Reply:
We could make it clearer and use something like `add_cmd` to indicate that it will be replace by the relevant command for the given shell. Good point.

Reply:
@BassemNKhalil would you be able to change the 
`Add` with `add_cmd`
`Remove` with `remove_cmd`
`copy` with `copy_cmd` in the rules definitions and also to string replacing based on those tokens.

Reply:
All done - I've also added create_dir_cmd to the mix.